src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -95,82 +105,218 @@ public static int IndexOf(ref byte searchSpace, byte value, int length)
             return -1;
         }
 
-        public static bool SequenceEqual(ref byte first, ref byte second, int length)
+        public unsafe static bool SequenceEqual(ref byte first, ref byte second, int length)
         {
             Debug.Assert(length >= 0);
+            var isMatch = true;
 
-            if (Unsafe.AreSame(ref first, ref second))
-                return true;
-
-            int index = 0;
-            while (length >= 8)
+            if (length == 0)
             {
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
+                goto exit;
+            }
 
-                length -= 8;
+            fixed (byte* pFirst = &first)
+            fixed (byte* pSecond = &second)
+            {
+                var a = pFirst;
+                var b = pSecond;
+
+                if (a == b)
+                {
+                    goto exitFixed;
```

Comment:
Bloats the asm by inserting all the undo fixed asm and stack pops in place, for a rare case (both terms are the identical memory location); goto keeps it nice and tight. Using goto out of fixed equally does a lot of code bloat; which is why two labels. I see this as being a very hot method so trying to minimise instructions.

Will post before and after asm.